src/apify_client/_utils.py

```diff
@@ -122,3 +125,51 @@ def _catch_not_found_or_throw(exc: ApifyApiError) -> None:
         raise exc
 
     return None
+
+
+def _encode_json_to_base64(data: JSONSerializable) -> bytes:
+    """Encode JSON serializable data to base64 string."""
+    return base64.b64encode(json.dumps(data).encode("utf-8"))
+
+
+def _decode_base64_json(encoded_data: bytes) -> JSONSerializable:
+    """Decode base64 string into JSONSerializable data.
+
+    >>> _decode_base64_json(_encode_json_to_base64(1))
+    1
+    >>> _decode_base64_json(_encode_json_to_base64(1.1))
+    1.1
+    >>> _decode_base64_json(_encode_json_to_base64("apify"))
+    'apify'
+    >>> _decode_base64_json(_encode_json_to_base64(True))
+    True
+    >>> _decode_base64_json(_encode_json_to_base64(None)) is None
+    True
+    >>> _decode_base64_json(_encode_json_to_base64([1, 2, 3]))
+    [1, 2, 3]
+    >>> _decode_base64_json(_encode_json_to_base64({"apify": "rocks"}))
+    {'apify': 'rocks'}
+    """
+    return cast(JSONSerializable, json.loads(base64.b64decode(encoded_data).decode("utf-8")))
+
+
+def _filter_out_none_values(dictionary: Dict) -> Dict:
+    """Return copy of the dictionary, omitting all keys for which values are None.
+
+    >>> _filter_out_none_values({"k1": "v1", "k2": None})
+    {'k1': 'v1'}
+    """
+    return {k: v for k, v in dictionary.items() if v is not None}
+
+
+def _filter_out_none_values_recursively(dictionary: Dict) -> Dict:
+    """Return copy of the dictionary, recursivelhy omitting all keys for which values are None.
+
+    >>> _filter_out_none_values_recursively({"k1": "v1", "k2": None, "k3": {"k4": "v4", "k5": None}})
+    {'k1': 'v1', 'k3': {'k4': 'v4'}}
+    """
+    return {
+        k: v if not isinstance(v, Dict) else _filter_out_none_values_recursively(v)
```

Comment:
Recursion is usually pretty unsafe to use (due to the possibility of exceeding max call stack depth) and incurs the unnecessary overhead of a function call. It's not really a problem here due to the small scale of the intended dictionaries, of course. It is more a matter of principle.

Reply:
Good point, but I don't really think it should be a problem here. Do you have strong opinion on this? I guess we can rewrite it using queue / stack, but I don't think it's really needed.

Reply:
I think we can keep the recursion here, since it should never become a problem. Just be careful where you use the function.